index.js

```diff
@@ -1,69 +1,65 @@
-var fs = require('fs')
-var path = require('path')
-var mkdirp = require('mkdirp')
-var includePathSearcher = require('include-path-searcher')
-var CachingWriter = require('broccoli-caching-writer')
-var less = require('less')
-var merge = require('lodash-node/modern/object/merge')
+var fs = require('fs');
+var path = require('path');
+var mkdirp = require('mkdirp');
+var includePathSearcher = require('include-path-searcher');
+var CachingWriter = require('broccoli-caching-writer');
+var less = require('less');
+var merge = require('lodash-node/modern/object/merge');
 var RSVP = require('rsvp');
 var writeFile = RSVP.denodeify(fs.writeFile);
 
 module.exports = LessCompiler;
 
-LessCompiler.prototype = Object.create(CachingWriter.prototype)
-LessCompiler.prototype.constructor = LessCompiler
+LessCompiler.prototype = Object.create(CachingWriter.prototype);
+LessCompiler.prototype.constructor = LessCompiler;
 
 function LessCompiler (sourceTrees, inputFile, outputFile, options) {
   if (!(this instanceof LessCompiler)) {
     return new LessCompiler(sourceTrees, inputFile, outputFile, options)
   }
 
-  CachingWriter.apply(this, [arguments[0]].concat(arguments[3]))
+  CachingWriter.apply(this, [arguments[0]].concat(arguments[3]));
 
-  options = options || {};
-  if (options.sourceMap) {
-    if (typeof options.sourceMap !== 'object') {
-      options.sourceMap = {};
-    }
-    if (!options.sourceMap.sourceMapURL) {
-      options.sourceMap.sourceMapURL = outputFile + '.map';
-    }
-  }
-
-  this.sourceTrees = sourceTrees
-  this.inputFile = inputFile
-  this.outputFile = outputFile
-  this.lessOptions = options
+  this.sourceTrees = sourceTrees;
+  this.inputFile   = inputFile;
+  this.outputFile  = outputFile;
+  this.options     = options || {};
 }
 
 LessCompiler.prototype.updateCache = function (srcDir, destDir) {
-  var destFile = destDir + '/' + this.outputFile
+  var destFile = destDir + '/' + this.outputFile;
 
   mkdirp.sync(path.dirname(destFile));
 
-  var lessOptions = {
-    filename: includePathSearcher.findFileSync(this.inputFile, srcDir),
-    paths: srcDir
-  }
+  var options = this.options;
 
-  merge(lessOptions, this.lessOptions)
+  options.filename = includePathSearcher.findFileSync(this.inputFile, srcDir);
 
-  lessOptions.paths = [path.dirname(lessOptions.filename)].concat(lessOptions.paths);
-  data = fs.readFileSync(lessOptions.filename, 'utf8');
+  options.paths = [path.dirname(options.filename)].concat(options.paths);
 
-  return less.render(data, lessOptions).
+  if (options.sourceMap) {
+    if (typeof options.sourceMap !== 'object') {
+      options.sourceMap = {};
+    }
+
+    options.sourceMap.sourceMapURL = this.outputFile + '.map';
```

Comment:
Previously you were able to pass a sourceMapURL (line 29).  So I'm slightly confused or misunderstanding the issue.